File: Traitement/Annales/exercice3.py
import pandas as pd
from pprint import pprint

df = pd.read_csv('Annales/fete-de-la-musique-2019.csv', sep=';')

# Filter concerts with age minimum >= 18
# fillna(0) replaces NaN values with 0
# astype(int) converts the "Age minimum" column to integers
# Filter the DataFrame to get rows where "Age minimum" is >= 18
adult_concerts = df[df["Age minimum"].fillna(0).astype(int) >= 18]
for i, concert in adult_concerts.iterrows():
    print(i, " ", concert["Titre (fr)"])

# Count concerts by region
# dropna() removes NaN values from the "Région" column
# value_counts() counts the occurrences of each region
dictRegion = df["Région"].dropna().value_counts().to_dict()

# Count labels
# dropna() removes NaN values from the "label_multivalued" column
# str.split(';') splits the labels into lists
# explode() transforms these lists into separate rows
# value_counts() counts the occurrences of each label
dictLabels = df["label_multivalued"].dropna().str.split(';').explode().value_counts().to_dict()

print(dictLabels.keys())
pprint(dictRegion)
pprint(dictLabels)


import pandas as pd
import matplotlib.pyplot as plt

# Chargement des données
df = pd.read_csv("data.tsv", sep="\t")

# Liste des pays présents dans les données
pays_list = df["pays"].unique()

# Création des figures
fig, axes = plt.subplots(1, 2, figsize=(14, 6))

# Fonction pour tracer un boxplot avec Matplotlib
def boxplot(ax, data, column, title, ylabel):
    grouped_data = [data[data["pays"] == pays][column].dropna() for pays in pays_list]
    ax.boxplot(grouped_data, labels=pays_list, patch_artist=True)
    ax.set_title(title)
    ax.set_xlabel("Pays")
    ax.set_ylabel(ylabel)
    ax.tick_params(axis='x', rotation=90)

# Boxplot pour la répartition des âges
boxplot(axes[0], df, "âge", "Répartition des âges par pays", "Âge")

# Boxplot pour la répartition des tailles
boxplot(axes[1], df, "taille", "Répartition des tailles par pays", "Taille (cm)")

# Sauvegarde et affichage du graphique
plt.tight_layout()
plt.savefig("repartition_ages_tailles.png")
plt.show()

File: Traitement/Annales/exercice3.R
# Read the CSV file into a dataframe
concerts <- read.csv("fete-de-la-musique-2019.csv", sep = ";", 
                     stringsAsFactors = FALSE)

# Select relevant columns
concerts <- concerts[, c("Titre..fr.", "Age.minimum", "Région", 
                         "label_multivalued")]

# Initialize dictionaries to count regions and labels
dict_region <- list()
dict_labels <- list()

# Iterate over each concert
for (i in seq_len(nrow(concerts))) {
  concert <- concerts[i, ]
  
  # Check if the concert is for adults (age >= 18)
  age_minimum <- as.numeric(concert["Age.minimum"])
  if (!is.na(age_minimum) && age_minimum >= 18) {
    print(paste(i, concert["Titre..fr."]))
  }
  
  # Count the number of concerts per region
  region <- concert[["Région"]]
  if (!is.na(region) && region != "NA") {
    if (!(region %in% names(dict_region))) {
      dict_region[[region]] <- 0
    }
    dict_region[[region]] <- dict_region[[region]] + 1
  }
  
  # Count the number of concerts per label
  labels <- concert[["label_multivalued"]]
  if (!is.na(labels)) {
    labels <- unlist(strsplit(labels, ";"))
    for (label in labels) {
      if (!(label %in% names(dict_labels))) {
        dict_labels[[label]] <- 0
      }
      dict_labels[[label]] <- dict_labels[[label]] + 1
    }
  }
}

# Filter out empty regions
dict_region <- Filter(function(x) x != "", dict_region)
dict_region <- dict_region[names(dict_region) != ""]

# Print the results
print(names(dict_labels))
print(dict_region)
print(dict_labels)



# Chargement des bibliothèques
library(ggplot2)
library(readr)

# Lecture du fichier TSV
df <- read_tsv("data.tsv")

# Création des graphiques
p1 <- ggplot(df, aes(x = pays, y = âge)) +
  geom_boxplot() +
  labs(title = "Répartition des âges par pays", x = "Pays", y = "Âge") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

p2 <- ggplot(df, aes(x = pays, y = taille)) +
  geom_boxplot() +
  labs(title = "Répartition des tailles par pays", x = "Pays", y = "Taille (cm)") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

# Sauvegarde en image
ggsave("repartition_ages_tailles.png", arrangeGrob(p1, p2, ncol = 2), width = 14, height = 6)

